### ISSUE TITLE
HTTPS/SSL not working

### ISSUE THREAD

**Issue by fahadakhan96** [zodern/meteor-up#802]
The following is my `mup.js`:
```
module.exports = {
  servers: {
    one: {
      // TODO: set host address, username, and authentication method
      host: '<hostname>',
      username: '<username>',
      pem: '<path to pem>',
      password: '<password>'
      // or neither for authenticate from ssh-agent
    }
  },

  app: {
    // TODO: change app name and path
    name: '<name>',
    path: '<path to app folder>',

    servers: {
      one: {},
    },

    buildOptions: {
      serverOnly: true,
    },

    env: {
      // TODO: Change to your app's url
      // If you are using ssl, it needs to start with https://
      ROOT_URL: 'https://<hostname>',
      MONGO_URL: 'mongodb://localhost/meteor',
    },

    ssl: { // (optional)
      // Enables let's encrypt (optional)
      autogenerate: {
        email: '<my_personal_email>',
        // comma separated list of domains
        domains: '<hostname>'
      }
    },

    docker: {
      // change to 'abernix/meteord:base' if your app is using Meteor 1.4 - 1.5
      image: 'abernix/meteord:node-8.4.0-base',
    },

    // Show progress bar while uploading bundle to server
    // You might need to disable it on CI servers
    enableUploadProgressBar: true
  },

  mongo: {
    port: 27017,
    version: '3.4.1',
    servers: {
      one: {}
    }
  }
};
```
I'm running Windows whereas the server is an AWS EC2 Ubuntu instance. Both `mup.cmd setup` and `mup.cmd deploy` complete successfully. However I can only access my app through `http` and not `https`. What am I missing?

**Comment by jonataw** [comment]
Facing this issue myself. The problem is **TOS hash mismatch**. Let's encrypt ToS changed today (https://letsencrypt.org/documents/2017.11.15-LE-SA-v1.2.pdf) which changed the hash.

It seems like we depend on [simp_le](https://github.com/zenhack/simp_le/) which uses a hardcoded hash. The hash has now been updated and will be fetched from API in the future: https://github.com/zenhack/simp_le/commit/adec4b99366fd9a67b2650ec64197e2a13bc12b3

Still not sure myself how to proceed from here.

**Comment by bvanderdrift** [comment]
Right, https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion/pull/282 is open and we should wait for it.

Might you want to fix it right now, you should make some manual changes in the code that `mup` uploads.

First: run on your own machine
`mup deploy`

then ssh into your server, and edit the file `/opt/<AppName>/config/start.sh`

Replace the line:

`sudo docker pull jrcs/letsencrypt-nginx-proxy-companion:$LETS_ENCRYPT_VERSION`

with:

`git clone --branch tos-fix https://github.com/buchdag/docker-letsencrypt-nginx-proxy-companion.git`
`sudo cd docker-letsencrypt-nginx-proxy-companion`
`docker build -t jrcs/letsencrypt-nginx-proxy-companion:latest ..`
`cd ..`

then exit the ssh session and run

`mup start` and NOT the `mup deploy` command. Running deploy will reset the start script.

Thanks to @buchdach for fixing the issue. Let's hope the PR gets accepted soon.

**Comment by buchdag** [comment]
@bvanderdrift I don't know if the expected value of `LETS_ENCRYPT_VERSION` in your script is `latest` or not, so you might want to do

`docker build -t jrcs/letsencrypt-nginx-proxy-companion:$LETS_ENCRYPT_VERSION .`

instead.

Also be careful, the build command should end with only one `.` (ie build in the current working dir), not two.

**Comment by bvanderdrift** [comment]
@buchdag ooh you're right. It's latest but using the var is better. In the context of the script though the image should be stored in the parent folder, since that's where the start.sh script expects it.

**Comment by buchdag** [comment]
The trailing `.` don't tell docker where to build **to** but where to build **from**, ie where to expect a `Dockerfile` and where the build context root will be if no further arguments are passed.

Neither `docker build` nor `docker pull` put images in the current working directory, they both go to the same Docker Root Dir (`/var/lib/docker` by default) , under the `image` subfolder.

**Comment by zodern** [comment]
Running `mup restart` or `mup proxy start` will pull the latest version of `jrcs/letsencrypt-nginx-proxy-companion` which has the fix.

**Comment by wildhart** [comment]
I'm still getting this TOS mismatch.  I only installed mup yesterday so should have the latest version of everything.  I also manually pulled the latest version of `jrcs/letsencrypt-nginx-proxy-companion` and that didn't help.

```
TOS hash mismatch. Found: cc88d8d9517f490191401e7b54e9ffd12a2b9082ec7a1d4cec6101f9f1647e7b.
```

I'm using mup 1.3.5 (downgraded from 1.3.6 due to #816).  I'm using the `proxy.ssl` method because I couldn't get the `meteor.ssl` method to work at all.
